github.go

```diff
@@ -0,0 +1,100 @@
+package main
+
+import (
+	"net/http"
+	"fmt"
+	"strings"
+	"bytes"
+	"encoding/json"
+)
+
+type gitHubRepo struct {
+	Owner string // The GitHub account name under which the repo exists
+	Name  string // The GitHub repo name
+}
+
+// Modeled directly after the api.github.com response
+type gitHubTagsApiResponse struct {
+	Name       string // The tag name
+	ZipBallUrl string // The URL where a ZIP of the release can be downloaded
+	TarballUrl string // The URL where a Tarball of the release can be downloaded
+	Commit     gitHubTagsCommitApiResponse
+}
+
+// Modeled directly after the api.github.com response
+type gitHubTagsCommitApiResponse struct {
+	Sha string // The SHA of the commit associated with a given tag
+	Url string // The URL at which additional API information can be found for the given commit
+}
+
+// Fetch all tags from the given GitHub repo
+func FetchTags(githubRepoUrl string, githubToken string) ([]string, *fetchError) {
+	repo, err := ExtractUrlIntoGitHubRepo(githubRepoUrl)
+	if err != nil {
+		return []string{}, newErr(err)
+	}
+
+	// Make an HTTP request, possibly with the gitHubOAuthToken in the header
+	httpClient := &http.Client{}
+	req, err := http.NewRequest("GET", fmt.Sprintf("https://api.github.com/repos/%s/%s/tags", repo.Owner, repo.Name), nil)
+	if err != nil {
+		return []string{}, newErr(err)
+	}
+
+	if githubToken != "" {
+		req.Header.Set("Authorization", fmt.Sprintf("token %s", githubToken))
+	}
```

Comment:
Although the code is isomorphic, the URL is different in each case and requires different params. I split out into dedicated functions, though to at least take advantage of the isomorphism.